### PR TITLE
Pack all units

### PR DESCRIPTION
**Pull request by ebetica**
Is there any reason we don't pack units for all players in non-replays? @jgehring

### CODE REVIEW

**Comment by jgehring** [comment]
IIRC historically, we only packed a subset of units, and those subsets had different criteria for ally/enemy/neutral. Yeah, we can just pack all.

File: BWEnv/src/controller.cc [hunk 1]
```diff
@@ -984,33 +981,17 @@ void Controller::packCreep(replayer::Frame& f) {
   }
 }
 
-void Controller::packMyUnits(replayer::Frame& f) {
-  for (auto& u : BWAPI::Broodwar->self()->getUnits()) {
-    // Ignore the unit if it no longer exists
+void Controller::packUnits(replayer::Frame& f) {
+  for (auto& u : BWAPI::Broodwar->getAllUnits()) {
+    if (u->getPlayer() == nullptr)
+    continue;
```
Comment (jgehring) [inline BWEnv/src/controller.cc]: Nit indent

File: BWEnv/src/controller.cc [hunk 2]
```diff
@@ -984,33 +981,17 @@ void Controller::packCreep(replayer::Frame& f) {
...
-  }
-}
-
-void Controller::packNeutral(replayer::Frame& f) {
-  for (auto& u : BWAPI::Broodwar->getNeutralUnits()) {
-    addUnit(
-        u,
-        f,
-        BWAPI::Broodwar->neutral()); // TODO: only when the state changes
+    addUnit(u, f, u->getPlayer()); // TODO: only when the state changes
```
Comment (jgehring) [inline BWEnv/src/controller.cc]: Let's remove this TODO? This is taken care of by frame diffs.

**Review by jgehring — APPROVED** [review]

File: BWEnv/src/controller.cc [hunk 3]
```diff
@@ -984,33 +981,17 @@ void Controller::packCreep(replayer::Frame& f) {
   }
 }
 
-void Controller::packMyUnits(replayer::Frame& f) {
-  for (auto& u : BWAPI::Broodwar->self()->getUnits()) {
-    // Ignore the unit if it no longer exists
+void Controller::packUnits(replayer::Frame& f) {
+  for (auto& u : BWAPI::Broodwar->getAllUnits()) {
+    if (u->getPlayer() == nullptr)
```
Comment (dgant) [inline BWEnv/src/controller.cc]: When do we expect this to be true?

**Pull request merged**